cypress/plugins/index.js

```diff
@@ -1,13 +1,110 @@
 const getChallenge = require('../../curriculum/getChallenges');
+const env = require('../../config/env.json');
 
 module.exports = on => {
   on('task', {
     // return an array of all challenge paths
     //  superBlock, block,  this are unused variables in the condition Husky doesn't like that
-    getAllChallengePaths({ lang }) {
-      const curriculum = getChallenge.getChallengesForLang(lang);
+    getCurriculum({ lang }) {
+      let curriculum;
 
-      return curriculum['responsive-web-design'];
+      if (!curriculum) {
+        curriculum = getChallenge.getChallengesForLang(lang);
+      }
+
+      return curriculum;
+    },
+
+    scopeCurriculum({ curriculum, superblock, block, challenge }) {
+      let challengePaths = [];
+
+      if (superblock && block === null) {
+        // Get blocks in superblock
+        block = Object.keys(curriculum[superblock]['blocks']);
+
+        // Get challenges in block
+
+        block.forEach(blocks => {
+          const upcommingChange =
+            curriculum[superblock]['blocks'][blocks]['meta'][
+              'isUpcomingChange'
+            ];
+
+          // Check if block is upcoming change
+          if (!upcommingChange && !env.showUpcomingChanges) {
+            let superBlockPath = superblock;
+            let blockPath =
+              curriculum[superblock]['blocks'][blocks]['meta']['dashedName'];
+            let challengeArr =
+              curriculum[superblock]['blocks'][blocks]['challenges'];
+
+            challengeArr.forEach(challengePath => {
+              challengePaths.push(
+                `/learn/${superBlockPath}/${blockPath}/${challengePath['dashedName']}`
+              );
+            });
+          } else if (
+            upcommingChange ||
+            (!upcommingChange && env.showUpcomingChanges)
```

Comment:
```suggestion
            upcommingChange && env.showUpcomingChanges
```
We can save a few characters here.